### DISCUSSION TITLE
Multi-step Form & JS

### DISCUSSION THREAD

**Discussion by acienfuegos** [Budibase/budibase#13275] [Help] [answered]
Hi!

I found very interesting the multi-step form block.

But I found issues when I'm trying to jump with a conditional step.

In our case, with JS (I'm not sure if the code is correct) 

Maybe I'm using a wrong approach for this, but I will explain deeper our case, with several screenshots.

First step, 1, has the following data:

![image](https://github.com/Budibase/budibase/assets/13641897/bea3bdc9-6e10-4262-86cb-a9c3f4b77533)

Nothing special, but when we are moving to step 2, we need to move forward to step 3 or 4, if the number of days are different.

![image](https://github.com/Budibase/budibase/assets/13641897/a41ed612-bef6-4f18-8883-d1010cc24b5f)

Talking about button: Next, just configured it:

![image](https://github.com/Budibase/budibase/assets/13641897/26ef4393-fd6c-422e-86b4-2596d2dbe5c2)

And tried this:

![image](https://github.com/Budibase/budibase/assets/13641897/678e99dd-b1f2-4616-b34a-03b014393b71)

And this:

<img width="487" alt="Captura de pantalla 2024-03-18 a las 13 49 51" src="https://github.com/Budibase/budibase/assets/13641897/54149f9e-a082-4df9-a1dd-9df27bb69881">

But nothing happened... we are a bit stuck, and as I mentioned maybe we are in a wrong approach to do this.

The idea, is have for example step 1 and 2 common.

But it depends you choose in step 2, you're move directly to step 3 or step 4.

Thanks in advance!

**Comment by mikesealey** [comment, marked as answer]
Hi @acienfuegos I've been looking into this for you. I think your logic is fine, but it might just be a little typo in your `if` condition.

Your condition mentions the variable `$(Multi-step Form Block 1.Fields.Number of days)`, but if you take a close look at the closing brackets `)` you're doing the comparison `== "1"` a little too soon. The first closing bracket is the end of the variable, then second is the end of your `if` condition.

You might also have the return statement a little off - it's asking for simply a number, so you can just return a number, not the variable

Please try

```
if ($(Multi-step Form Block 1.Fields.Number of days) === "1") {
   return 3
} else {
   return 4
}
```

Here's a screenshot of the condition I used to arrive at this solution
![image](https://github.com/Budibase/budibase/assets/39269380/3c5894bc-a30f-4e76-b3ae-072c81e2405a)

And here's an export of the app I used to play around with this scenario to find the answer for you.
[13275-export-1713891565511.tar.gz](https://github.com/Budibase/budibase/files/15080268/13275-export-1713891565511.tar.gz)

If you want to import my app to have a closer look, you can use [This page](https://docs.budibase.com/docs/export-and-import-apps#import-an-application) of the documentation for help.

**Reply by acienfuegos** [reply]
Thanks for your answer @mikesealey !

I will try to review ASAP, and of course I will come back if it works!

Thanks again! 👌

**Reply by acienfuegos** [reply]
Hi there @mikesealey!

Just tested, and it works like a charm 👌 

Now I need to play how to save the data once... because I'm doing twice (just found where) but need to understand how to keep the data for just one day or 2, not duplicated.

Maybe I will come back here to ask 😄 

Thanks a lot again!
